qermit/zero_noise_extrapolation/zne.py

```diff
@@ -139,10 +158,14 @@ def gate(circ: Circuit, noise_scaling: float, **kwargs) -> Circuit:
             command_circ_dict.update({"commands": [command]})
             command_circ = Circuit().from_dict(command_circ_dict)
 
-            # Find the inverse of the command
-            inverse_command_circ = command_circ.dagger()
-            inverse_command_circ_dict = inverse_command_circ.to_dict()
-            inverse_command = inverse_command_circ_dict["commands"]
+            # Commands which are not to be folded may not be invertible (for
+            # example barriers) and so the inverse is not calculated in that
+            # case.
+            if num_folds[command_index] > 0:
+                # Find the inverse of the command
```

Comment:
Not important now, but this whole process of replacing single commands in c_dict could probably do with a bit more explanation in the comments

Reply:
Thanks, I'll keep that in mind. I'm thinking of doing a small overhall here, when I have a little more time. I'll remember this comment when I do that.